src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs

```diff
@@ -131,32 +134,35 @@ protected virtual async Task<TestBlockchain> Build(ISpecProvider? specProvider =
         TrieStore = new TrieStore(StateDb, LogManager);
         State = new WorldState(TrieStore, DbProvider.CodeDb, LogManager, new PreBlockCaches());
 
-        // Eip4788 precompile state account
-        if (specProvider?.GenesisSpec?.IsBeaconBlockRootAvailable ?? false)
+        if (!keepStateEmpty)
```

Comment:
Instead of adding a parameter maybe better to:
1. Put below state modifying code into virtual method and then override it and do nothing.
2. Have a property That indicates if state should be empty - set it or override it in usages.